autoload/codeium.vim

```diff
@@ -42,49 +42,37 @@ function! codeium#Accept() abort
   let default = get(g:, 'codeium_tab_fallback', pumvisible() ? "\<C-N>" : "\t")
 
   let current_completion = s:GetCurrentCompletionItem()
-  if current_completion isnot v:null
-    let range = current_completion.range
-    let start_position = get(range, 'startPosition', {})
-    let end_position = get(range, 'endPosition', {})
-    let start_row = get(start_position, 'row', 0) + 1
-    let start_col = get(start_position, 'col', 0) + 1
-    let end_row = get(end_position, 'row', 0) + 1
-    let end_col = get(end_position, 'col', 0) + 1
-    let suffix = get(current_completion, 'suffix', {})
-    let suffix_text = get(suffix, 'text', '')
-    let delta = get(suffix, 'deltaCursorOffset', 0)
-
-    let text = current_completion.completion.text . suffix_text
-    if empty(text)
-      return default
-    endif
+  if current_completion is v:null
+    return default
+  endif
 
-    let s:completion_text = text
+  let range = current_completion.range
+  let suffix = get(current_completion, 'suffix', {})
+  let suffix_text = get(suffix, 'text', '')
+  let delta = get(suffix, 'deltaCursorOffset', 0)
+  let start_offset = get(range, "startOffset", 0)
+  let end_offset = get(range, "endOffset", 0)
 
-    let insert_text = "\<C-R>\<C-O>=codeium#CompletionText()\<CR>"
-    let move_to_start = "\<C-O>:call cursor(" . start_row . ',' . start_col . ")\<CR>"
-    let move_to_end = "\<C-O>:call cursor(" . end_row . ',' . end_col . ")\<CR>"
+  let text = current_completion.completion.text . suffix_text
+  if empty(text)
+    return default
+  endif
+  
+  let delete_range = ""
+  if end_offset - start_offset > 0
+    let delete_range = " \<Esc>x0d" . (end_offset - start_offset) . "li"
```

Comment:
```suggestion
      " We insert a space, escape to normal mode, then delete the inserted space.
      " This lets us "accept" any auto-inserted indentation which is otherwise
      " removed when we switch to normal mode.
    let delete_range = " \<Esc>x0d" . (end_offset - start_offset) . "li"
```